src/pairs/GameState.py

```diff
@@ -17,11 +18,37 @@ def deepcopy(self):
         c = GameState(self.deck.deepcopy(), self.target_score, self.players.deepcopy())
         return c
 
-    def draw_next(self):
-        pass
+    def draw_next(self, game):
+        curr_state = game.deepcopy()
+        new_state = curr_state.deepcopy()
+        states = []
+        possible = []
+        for card in curr_state.curr_deck:
+            if curr_state.curr_deck[card] != 0:
+                possible.append(card)
+        for i in range(len(possible)):
+            new_state.players[0].add_card(curr_state.deck.draw_specific_card(i))
+            states.append(new_state)
+            new_state = curr_state.deepcopy()
+        return states
+
+    def fold_next(self, game):
+        curr_state = game.deepcopy()
+        minimum = min([player.points for player in game.players])
+        game.players[0].points += minimum
+        for player in game.players:
+            game.deck.add_dict_discards(player.hand_state)
+            player.reset()
+        curr_state.players.rotate(1)
+        return curr_state
 
-    def fold_next(self):
-        pass
+    def ending_state(self):
```

Comment:
A little bit of a weird scenario that probably shouldn't happen but you should safeguard code against it may be if there are multiple simultaneous losers (multiple people have scores greater than the target score). In this case, each player with scores greater than or equal to the target score will get probabilities 0 and everyone else will get probability 1.
Some pythonic ways to write this:
```
if any(player.points >= self.target_score for player in self.players):
    return np.array([float(player.points < self.target_score) for player in self.players])
return None
```

Reply:
done